src/ophys_etl/modules/suite2p_registration/suite2p_utils.py

```diff
@@ -0,0 +1,116 @@
+from suite2p.registration.rigid import (apply_masks, compute_masks, phasecorr,
+                                        phasecorr_reference, shift_frame)
+
+
+def load_initial_frames(file_path: str,
+                        h5py_key: str,
+                        nimg_init: int) -> np.ndarray:
+    """Load a subset of frames from the hdf5 data specified by file_path.
+
+    Parameters
+    ----------
+    file_path : str
+        Location of the raw 2Photon, HDF5 data to load.
+    h5py_key : str
+        Name of the dataset to load from the HDF5 file.
+    nimg_init : int
+        Number of frames to load from the input HDF5 data.
+
+    Returns
+    -------
+    frames : array-like, (nimg_init, Ly, Lx)
+        Selected frames from the input raw data linearly spaced in index of the
+        time axis. If nimg_init > nframes, a number of frames equal to nframes
+        is returned.
+    """
+    with h5py.File(file_path, "r") as hdf5_file:
+        # Get a set of linear spaced frames to load from disk.
+        tot_frames = hdf5_file[h5py_key].shape[0]
+        requested_frames = np.linspace(0,
+                                       tot_frames,
+                                       1 + min(nimg_init, tot_frames),
+                                       dtype=int)[:-1]
+        frames = hdf5_file[h5py_key][requested_frames]
+    return frames
+
+
+def compute_reference(frames: np.ndarray,
+                      niter: int,
+                      maxregshift: float,
+                      smooth_sigma: float,
+                      smooth_sigma_time: float) -> np.ndarray:
+    """Computes a stacked reference image from the input frames.
+
+    Modified version of Suite2P's compute_reference function with no updating
+    of input frames. Picks initial reference then iteratively aligns frames to
+    create reference. Does not reproduce the Suite2p 1Photon code path.
+
+    Parameters
+    ----------
+    frames : array-like, (nimg_init, Ly, Lx)
+        Set of frames to create a reference from.
+    niter : int
+        Number of iterations to perform when creating the reference image.
+    maxregshift : float
+        Maximum shift allowed as a fraction of the image width or height, which
+        ever is longer.
+    smooth_sigma : float
+        Width of the Gaussian used to smooth the phase correlation between the
+        reference and the frame with which it is being registered.
+    smooth_sigma_time : float
+        Width of the Gaussian used to smooth between multiple frames by before
+        phase correlation.
+
+    Returns
+    -------
+    refImg : array-like, (Ly, Lx)
+        Reference image created from the input data.
+    """
+    # Get initial reference image from suite2p.
+    ref_image = pick_initial_reference(frames)
+
+    for idx in range(niter):
+        # Compute the number of frames to select in creating the reference
+        # image.
+        nmax = int(frames.shape[0] * (1. + idx) / (2 * niter))
```

Comment:
I would removed the magic number 2 (and maybe 1.?). Set it in a descriptive constant.

Reply:
This was inherited from suite2p as a "magic number" and will likely go away if we end up deciding to work on #394. As it's a little more algorithmic (the goal is to select half the data at most) would a extra sentence in the above comment to the effect of "At most we select half to the input images." suffice?

Reply:
Yes, that is fine.